### PR TITLE
Allow protocol-specific parameters to be inherited on servers

### PR DESCRIPTION
**Pull request by olix0r**
Currently, only a few non-protocol-specific parameters are shared from a router
to its servers.  This makes it impossible to set router-levelcodec parameters
(e.g. MaxChunkSize).

Add a `configureServer` method to ProtocolInitializer so that each protocol may
provide configuration-sharing logic as needed.  By default, we share HTTP codec
parameters from routers to servers.

<hr />
Manually tested locally: uploading a 38M file fails with a 413 unless `maxRequestKB` is set appropriately.

### CODE REVIEW

File: linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/HttpConfig.scala [hunk 1]
```diff
@@ -37,6 +37,21 @@ class HttpInitializer extends ProtocolInitializer.Simple {
       .configured(RoutingFactory.DstPrefix(Path.Utf8(name)))
   }
 
+  /**
+   * Apply the router's codec coonfiguration applies to the server.
+   */
+  override protected def configureServer(router: Router, server: Server): Server = {
+    println(s"configuring server with ${router.params}")
```
Comment (gtcampbell) [inline linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/HttpConfig.scala]: println

File: linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/HttpConfig.scala [hunk 2]
```diff
@@ -37,6 +37,21 @@ class HttpInitializer extends ProtocolInitializer.Simple {
       .configured(RoutingFactory.DstPrefix(Path.Utf8(name)))
   }
 
+  /**
+   * Apply the router's codec coonfiguration applies to the server.
```
Comment (gtcampbell) [inline linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/HttpConfig.scala]: typo: "coonfiguration" (and I think the "applies" may be extraneous, unless I misunderstand its intent?)

**Comment by gtcampbell** [comment]
:star: is it possible to write an automated test for this?

**Comment by olix0r** [comment]
whoops, thanks for those catches (that's what i get for pushing a fix from the bus ;p)

as for tests -- it's certainly possible to test but about 3x as much work as the fix.  I'll ry to get tests together...

**Comment by olix0r** [comment]
⭐  self-review

**Pull request merged**